File: crawler.py
from selenium import webdriver
import pickle
from bs4 import BeautifulSoup as bs

# read from configuration
chromedriver_path = '/webdriver/chromedriver'

# login with cookie
# reference: https://stackoverflow.com/questions/45417335/python-use-cookie-to-login-with-selenium
def save_cookie(browser, path):
    with open(path, 'wb') as filehandler:
        pickle.dump(browser.get_cookies(), filehandler)

def load_cookie(browser, path):
  # need to browser.refresh() after call this function
    with open(path, 'rb') as cookiesfile:
        cookies = pickle.load(cookiesfile)
        for cookie in cookies:
            browser.add_cookie(cookie)

def open_browser(mode=''):
  options = ''
  # headless mode
  # reference: https://stackoverflow.com/questions/16180428/can-selenium-webdriver-open-browser-windows-silently-in-background
  if mode is 'invisible':
    options = webdriver.ChromeOptions()
    options.add_argument("headless")
  browser = webdriver.Chrome(executable_path=chromedriver_path, chrome_options=options)
  browser.implicitly_wait(3)
  return browser

def login_naver(browser, userid, userpw):
  browser.get('https://nid.naver.com/nidlogin.login')
  browser.implicitly_wait(5)
  browser.find_element_by_name('id').send_keys(userid)
  browser.find_element_by_name('pw').send_keys(userpw)
  browser.find_element_by_xpath('//*[@id="frmNIDLogin"]/fieldset/input').click()

def check_login(browser):
  browser.switch_to_frame('minime')
  html = browser.page_source
  soup = bs(html, 'html.parser')
  print(soup.select('strong#user_name')[0])

File: crawler.cs
using OpenQA.Selenium;
using OpenQA.Selenium.IE;
using System.Threading;

/***************************************************
  Tips:
  (1) If the browser doesn't show up, enable protected mode for all zones in the security tab of Internet Options, and apply zoom 100%.
  (2) If SendKeys() is too slow, use 32-bit IEDriverServer.exe instead of 64-bit one.
****************************************************/

// path of IEDriverServer: %PROJECT_PATH%\bin\Debug
IWebDriver open_browser(string mode = "")
{
  var driverService = InternetExplorerDriverService.CreateDefaultService();
  driverService.HideCommandPromptWindow = true;
  IWebDriver browser = new InternetExplorerDriver(driverService); // PATH: %PROJECT_PATH%\bin\Debug
  if (mode.Equals("invisible")) browser.Manage().Window.Minimize();
  return browser;
}

void login_naver(IWebDriver browser, string userid, string userpw)
{
  browser.Url = "https://nid.naver.com/nidlogin.login";
  IWebElement login_id = browser.FindElement(By.Name("id"));
  IWebElement login_pw = browser.FindElement(By.Name("pw"));
  IWebElement login_button = browser.FindElement(By.XPath("//*[@id=\"frmNIDLogin\"]/fieldset/input"));
  login_id.Clear();
  login_pw.Clear();
  login_id.SendKeys(userid);
  login_pw.SendKeys(userpw);
  login_button.Click();
}

void check_login(IWebDriver browser)
{
  browser.SwitchTo().Frame("minime");
  IWebElement user_name = browser.FindElement(By.Id("user_name"));
  MessageBox.Show(user_name.Text);
}